### PR TITLE
Improving tests quality

### PR DESCRIPTION
**Pull request by kdybicz**
* Fixing issue with assertion in callback not taken into account while evaluating async tests outcome
* Fixing broken tests by changing wrong expected output format in conversion tests
* Adding new test cases, ie: to cover preserving and forcing change of aspect ratio while resizing
* Upgrading sinon to 1.17.7

### CODE REVIEW

File: test/resize-gif.js [hunk 1]
```diff
@@ -2,26 +2,42 @@
...
-const fsP          = pify(fs);
 
-test("Resize GIF with gifsicle", async t => {
-    const fixture  = fs.readFileSync(`${__dirname}/fixture/fixture.gif`);
-    const destPath = `${__dirname}/fixture/fixture_resized.gif`;
+let image;
+
+test.before(async t => {
+    const fixture = await pify(fs.readFile)(`${__dirname}/fixture/fixture.gif`);
+    image   = new ImageData("fixture/fixture.jpg", "fixture", fixture);
```
Comment (ysugimoto) [inline test/resize-gif.js]:
```
image   = new ImageData("fixture/fixture.gif", "fixture", fixture);
```

File: test/resize-gif.js [hunk 2]
```diff
@@ -2,26 +2,42 @@
...
-const fsP          = pify(fs);
 
-test("Resize GIF with gifsicle", async t => {
-    const fixture  = fs.readFileSync(`${__dirname}/fixture/fixture.gif`);
-    const destPath = `${__dirname}/fixture/fixture_resized.gif`;
+let image;
+
+test.before(async t => {
+    const fixture = await pify(fs.readFile)(`${__dirname}/fixture/fixture.gif`);
+    image   = new ImageData("fixture/fixture.gif", "fixture", fixture);
```
Comment (1000ch) [inline test/resize-gif.js]: Extra spaces.

File: test/resize-jpeg.js [hunk 3]
```diff
@@ -2,44 +2,72 @@
...
-test("Resize JPEG with cjpeg", async t => {
-    const fixture = await fsP.readFile(`${__dirname}/fixture/fixture.jpg`);
-    const destPath = `${__dirname}/fixture/fixture_resized_1.jpg`;
-    const resizer = new ImageResizer({size: 200});
-    const image   = new ImageData("fixture/fixture.jpg", "fixture", fixture);
+let image;
+
+test.before(async t => {
+    const fixture = await pify(fs.readFile)(`${__dirname}/fixture/fixture.jpg`);
+    image   = new ImageData("fixture/fixture.jpg", "fixture", fixture);
```
Comment (1000ch) [inline test/resize-jpeg.js]: Extra spaces.

File: test/resize-png.js [hunk 4]
```diff
@@ -2,44 +2,42 @@
...
-test("Resize PNG", async t => {
-    const fixture = await fsP.readFile(`${__dirname}/fixture/fixture.png`);
-    const destPath = `${__dirname}/fixture/fixture_resized_1.png`;
-    const resizer = new ImageResizer({size: 200});
-    const image = new ImageData("fixture/fixture.png", "fixture", fixture);
+let image;
+
+test.before(async t => {
+    const fixture = await pify(fs.readFile)(`${__dirname}/fixture/fixture.png`);
+    image   = new ImageData("fixture/fixture.png", "fixture", fixture);
```
Comment (1000ch) [inline test/resize-png.js]: Extra spaces.

**Comment by kdybicz** [comment]
https://blog.scottnonnenberg.com/top-ten-pull-request-review-mistakes/

**Comment by kdybicz** [comment]
Do not merge yet, I've learned about proper way of dealing with `pify` for `gm` and will present second PR soon.

**Comment by kdybicz** [comment]
Proposal of the alternative implementation of test changes https://github.com/ysugimoto/aws-lambda-image/pull/88/files Keep in mind that the failing tests because of something wrong in the code.

**Comment by kdybicz** [comment]
I'm closing this PR in favor of the https://github.com/ysugimoto/aws-lambda-image/pull/88

**Pull request closed without merging**